ChaosMod/Effects/db/Player/PlayerZoomZoomCam.cpp

```diff
@@ -0,0 +1,72 @@
+/*
+    Effect by Moxi, based on code from DrUnderscore & ProfessorBiddle
+*/
+
+#include <stdafx.h>
+static Camera zoomCamera = 0;
+float camZoom = 80.f;
+const float camZoomRate = 0.6f;
+int n = 0;
```

Comment:
This seems to be a flag whether it should decrease or increase
I would go another way:
* keep a variable which holds the amount it should change (already got the base with „ camZoomRate“)
* once you get below or above the limits -> negate the amount („* -1“)
* This way you don’t need to keep track of you’re increasing or decreasing and can bounds-check with one if and one negate statement

Reply:
That's so much better! Thank you for the suggestion. It's way cleaner now, 1 if instead of 4